src/daft-parquet/src/file.rs

```diff
@@ -371,9 +380,136 @@ impl ParquetFileReader {
 
     pub async fn read_from_ranges_into_table_stream(
         self,
-        _ranges: Arc<RangesContainer>,
-    ) -> BoxStream<'static, DaftResult<Table>> {
-        todo!("Implement streaming reads for remote parquet files")
+        ranges: Arc<RangesContainer>,
+        maintain_order: bool,
+        predicate: Option<ExprRef>,
+        original_columns: Option<Vec<String>>,
+        original_num_rows: Option<usize>,
+    ) -> DaftResult<BoxStream<'static, DaftResult<Table>>> {
+        let rt_handle = tokio::runtime::Handle::current();
+        let daft_schema = daft_core::schema::Schema::try_from(self.arrow_schema.as_ref())?;
+        let schema_ref = Arc::new(daft_schema);
+
+        let mut table_iters = Vec::with_capacity(self.row_ranges.len());
+        for row_range in self.row_ranges.iter() {
+            let rg = self
+                .metadata
+                .row_groups
+                .get(row_range.row_group_index)
+                .expect("Row Group index should be in bounds");
+            let num_rows = rg.num_rows().min(row_range.start + row_range.num_rows);
+            let columns = rg.columns();
+
+            let mut arr_iters = Vec::with_capacity(self.arrow_schema.fields.len());
+            for field in self.arrow_schema.fields.iter() {
+                let filtered_cols_idx = columns
```

Comment:
I believe so, but you should double check!

Reply:
Ok managed to refactor it so that it doesn't need idxs, just retrieves the column metadata directly for each field per rowgroup, which should be unique.